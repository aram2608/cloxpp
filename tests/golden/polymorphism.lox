class Animal {
  speak() { print "generic sound"; }
}

class Dog < Animal {
  speak() { print "woof"; }
}

class Cat < Animal {
  speak() { print "meow"; }
}

var d = Dog();
var c = Cat();
var a = Animal();

d.speak(); // expect "woof"
c.speak(); // expect "meow"
a.speak(); // expect "generic sound"