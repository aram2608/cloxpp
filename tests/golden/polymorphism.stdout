Token: CLASS class 1
Token: LEFT_BRACE { 1
Token: IDENTIFIER speak 2
Token: LEFT_PAREN ( 2
Token: RIGHT_PAREN ) 2
Token: LEFT_BRACE { 2
Token: PRINT print 2
Token: STRING "generic sound" 2
Token: SEMICOLON ; 2
Token: RIGHT_BRACE } 2
Token: RIGHT_BRACE } 3
Token: CLASS class 5
Token: LESS < 5
Token: LEFT_BRACE { 5
Token: IDENTIFIER speak 6
Token: LEFT_PAREN ( 6
Token: RIGHT_PAREN ) 6
Token: LEFT_BRACE { 6
Token: PRINT print 6
Token: STRING "woof" 6
Token: SEMICOLON ; 6
Token: RIGHT_BRACE } 6
Token: RIGHT_BRACE } 7
Token: CLASS class 9
Token: LESS < 9
Token: LEFT_BRACE { 9
Token: IDENTIFIER speak 10
Token: LEFT_PAREN ( 10
Token: RIGHT_PAREN ) 10
Token: LEFT_BRACE { 10
Token: PRINT print 10
Token: STRING "meow" 10
Token: SEMICOLON ; 10
Token: RIGHT_BRACE } 10
Token: RIGHT_BRACE } 11
Token: VAR var 13
Token: EQUAL = 13
Token: LEFT_PAREN ( 13
Token: RIGHT_PAREN ) 13
Token: SEMICOLON ; 13
Token: VAR var 14
Token: IDENTIFIER c 14
Token: EQUAL = 14
Token: LEFT_PAREN ( 14
Token: RIGHT_PAREN ) 14
Token: SEMICOLON ; 14
Token: VAR var 15
Token: IDENTIFIER a 15
Token: EQUAL = 15
Token: LEFT_PAREN ( 15
Token: RIGHT_PAREN ) 15
Token: SEMICOLON ; 15
Token: DOT . 17
Token: IDENTIFIER speak 17
Token: LEFT_PAREN ( 17
Token: RIGHT_PAREN ) 17
Token: SEMICOLON ; 17
Token: IDENTIFIER c 18
Token: DOT . 18
Token: IDENTIFIER speak 18
Token: LEFT_PAREN ( 18
Token: RIGHT_PAREN ) 18
Token: SEMICOLON ; 18
Token: IDENTIFIER a 19
Token: DOT . 19
Token: IDENTIFIER speak 19
Token: LEFT_PAREN ( 19
Token: RIGHT_PAREN ) 19
Token: SEMICOLON ; 19
Token: EOF  19
